src/pebble.py

```diff
@@ -276,6 +277,7 @@ def change_config(  # noqa: C901 pylint: disable=too-many-branches,too-many-stat
     """
     try:
         synapse.execute_migrate_config(container=container, charm_state=charm_state)
+        existing_synapse_config = _get_synapse_config(container)
         current_synapse_config = _get_synapse_config(container)
```

Comment:
isn't `existing_synapse_config` equal to `current_synapse_config`?, they are just one line apart